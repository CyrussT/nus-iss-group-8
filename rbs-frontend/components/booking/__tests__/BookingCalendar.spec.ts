import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import BookingCalendar from '../BookingCalendar.vue';

describe('BookingCalendar', () => {
  it('renders correctly', () => {
    const wrapper = mount(BookingCalendar, {
      global: {
        stubs: {
          UButton: true,
          UIcon: true
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
