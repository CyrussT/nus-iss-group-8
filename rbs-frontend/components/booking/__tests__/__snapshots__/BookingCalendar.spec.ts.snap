// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookingCalendar > renders correctly 1`] = `
"<div data-v-2958dc97="" class="rounded-lg divide-y divide-gray-200 dark:divide-gray-800 ring-1 ring-gray-200 dark:ring-gray-800 shadow bg-white dark:bg-gray-900 calendar-wrapper-card">
  <!--v-if-->
  <div class="px-4 py-5 sm:p-6">
    <div data-v-2958dc97="" class="calendar-wrapper">
      <div data-v-2958dc97="" attrs="[object Object]" class="resource-timeline fc fc-media-screen fc-direction-ltr fc-theme-standard fc-liquid-hack">
        <div class="fc-header-toolbar fc-toolbar fc-toolbar-ltr">
          <div class="fc-toolbar-chunk"><button type="button" title="prev" aria-pressed="false" class="fc-prev-button fc-button fc-button-primary"><span class="fc-icon fc-icon-chevron-left" role="img"></span></button></div>
          <div class="fc-toolbar-chunk">
            <h2 class="fc-toolbar-title" id="fc-dom-1">May 17, 2025</h2>
          </div>
          <div class="fc-toolbar-chunk"><button type="button" title="Next day" aria-pressed="false" class="fc-next-button fc-button fc-button-primary"><span class="fc-icon fc-icon-chevron-right" role="img"></span></button></div>
        </div>
        <div aria-labelledby="fc-dom-1" class="fc-view-harness fc-view-harness-passive">
          <div class="fc-resourceTimelineDay-view fc-view fc-resource-timeline fc-resource-timeline-flat fc-timeline fc-timeline-overlap-enabled">
            <table role="grid" class="fc-scrollgrid ">
              <colgroup>
                <col style="width: 20%;">
                <col>
                <col>
              </colgroup>
              <tbody role="rowgroup">
                <tr role="presentation" class="fc-scrollgrid-section fc-scrollgrid-section-header  fc-scrollgrid-section-sticky">
                  <th role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: hidden; overflow-y: hidden;">
                        <table role="presentation" class="fc-datagrid-header fc-scrollgrid-sync-table" style="width: 0px;">
                          <colgroup>
                            <col>
                          </colgroup>
                          <thead role="presentation">
                            <tr role="row">
                              <th role="columnheader" class="fc-datagrid-cell">
                                <div class="fc-datagrid-cell-frame" style="height: 0px;">
                                  <div class="fc-datagrid-cell-cushion fc-scrollgrid-sync-inner"><span class="fc-datagrid-expander fc-datagrid-expander-placeholder"><span class="fc-icon"></span></span><span class="fc-datagrid-cell-main">Resources</span></div>
                                </div>
                              </th>
                            </tr>
                          </thead>
                        </table>
                      </div>
                    </div>
                  </th>
                  <td role="presentation" class="fc-resource-timeline-divider fc-cell-shaded"></td>
                  <th role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: scroll; overflow-y: hidden;">
                        <div class="fc-timeline-header">
                          <table aria-hidden="true" class="fc-scrollgrid-sync-table" style="min-width: 720px; width: 0px;">
                            <colgroup>
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                              <col style="min-width: 30px;">
                            </colgroup>
                            <tbody>
                              <tr class="fc-timeline-header-row fc-timeline-header-row-chrono">
                                <th colspan="2" data-date="2025-05-17T07:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">7am</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T08:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">8am</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T09:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">9am</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T10:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">10am</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T11:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">11am</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T12:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">12pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T13:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">1pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T14:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">2pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T15:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">3pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T16:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">4pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T17:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-past">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">5pm</a></div>
                                </th>
                                <th colspan="2" data-date="2025-05-17T18:00:00" class="fc-timeline-slot fc-timeline-slot-label fc-day fc-day-sat fc-day-today fc-day-future">
                                  <div class="fc-timeline-slot-frame" style="height: 0px;"><a class="fc-timeline-slot-cushion fc-scrollgrid-sync-inner">6pm</a></div>
                                </th>
                              </tr>
                            </tbody>
                          </table>
                          <div class="fc-timeline-now-indicator-container"></div>
                        </div>
                      </div>
                    </div>
                  </th>
                </tr>
                <tr role="presentation" class="fc-scrollgrid-section fc-scrollgrid-section-body ">
                  <td role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: hidden; overflow-y: hidden;">
                        <table role="presentation" class="fc-datagrid-body fc-scrollgrid-sync-table" style="width: 0px;">
                          <colgroup>
                            <col>
                          </colgroup>
                          <tbody role="presentation"></tbody>
                        </table>
                      </div>
                    </div>
                  </td>
                  <td role="presentation" class="fc-resource-timeline-divider fc-cell-shaded"></td>
                  <td role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: scroll; overflow-y: hidden;">
                        <div class="fc-timeline-body " style="min-width: 720px;">
                          <div class="fc-timeline-slots">
                            <table aria-hidden="true" class="" style="min-width: 720px; width: 0px;">
                              <colgroup>
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                                <col style="min-width: 30px;">
                              </colgroup>
                              <tbody>
                                <tr>
                                  <td data-date="2025-05-17T07:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T07:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T08:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T08:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T09:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T09:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T10:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T10:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T11:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T11:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T12:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T12:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T13:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T13:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T14:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T14:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T15:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T15:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T16:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T16:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T17:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-past">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T17:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-future">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T18:00:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-major fc-slot fc-slot-sat fc-slot-today fc-slot-future">
                                    <div class=""></div>
                                  </td>
                                  <td data-date="2025-05-17T18:30:00" class="fc-timeline-slot fc-timeline-slot-lane fc-timeline-slot-minor fc-slot fc-slot-sat fc-slot-today fc-slot-future">
                                    <div class=""></div>
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                          <table aria-hidden="true" class="fc-scrollgrid-sync-table " style="min-width: 720px; width: 0px;">
                            <tbody></tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </td>
                </tr>
                <tr role="presentation" class="fc-scrollgrid-section fc-scrollgrid-section-footer  fc-scrollgrid-section-sticky">
                  <th role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: hidden; overflow-y: hidden;">
                        <div class="fc-scrollgrid-sticky-shim" style="width: 0px;"></div>
                      </div>
                    </div>
                  </th>
                  <td role="presentation" class="fc-resource-timeline-divider fc-cell-shaded"></td>
                  <th role="presentation">
                    <div class="fc-scroller-harness">
                      <div class="fc-scroller" style="overflow-x: auto; overflow-y: hidden;">
                        <div class="fc-scrollgrid-sticky-shim" style="min-width: 720px; width: 0px;"></div>
                      </div>
                    </div>
                  </th>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
